storagemarket/types/contract_deal_proposal_types.go

```diff
@@ -0,0 +1,33 @@
+package types
+
+import (
+	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/go-state-types/builtin/v9/market"
+	"github.com/ipfs/go-cid"
+)
+
+type ContractDealProposal struct {
+	PieceCID     cid.Cid
+	PieceSize    abi.PaddedPieceSize
+	VerifiedDeal bool
+	Client       address.Address
+
+	Label market.DealLabel
+
+	StartEpoch           abi.ChainEpoch
+	EndEpoch             abi.ChainEpoch
+	StoragePricePerEpoch abi.TokenAmount
+
+	ProviderCollateral abi.TokenAmount
+	ClientCollateral   abi.TokenAmount
+
+	Version string
+	Params  []byte
+}
+
+type ParamsVersion1 struct {
```

Comment:
nit: if ParamsVersion1 is specific to contracts, suggest we prefix it with Contract (or similar) to distinguish it from DealParams (used in libp2p deal proposals)